test/watch.js

```diff
@@ -0,0 +1,441 @@
+const test = require('ava');
+const fs = require('fs');
+const brunch = require('../lib');
+const {
+  prepareTestDir,
+  teardownTestDir,
+  fileContains,
+  fileDoesNotContains,
+  fileExists,
+  fileDoesNotExists,
+  requestBrunchServer
+} = require('./_test_helper');
+const fixturify = require('fixturify');
+
+var watcher;
+
+const watch = function(...args) {
+  watcher = brunch.watch(...args);
+};
+
+test.beforeEach.cb((t) => {
+  teardownTestDir();
+  prepareTestDir();
+
+  if (watcher) {
```

Comment:
I'm putting everything in `beforeEach` because in this ava version, if a test throws an error the `afterEach` is not called. This is why in the `beforeEach` first is called the `teardownTestDir` Because if in the previous run something failed, we cleanup the test state. Fortunately, ava master branch has a `afterEach.always` which was introduced [in this PR](https://github.com/sindresorhus/ava/pull/806), when it got a new release we can use it and test in a sane way.